loadsave.c

```diff
@@ -50,8 +54,23 @@ create_directory_listing(uint8_t *data)
 	*data++ = 'P';
 	*data++ = 'C';
 	*data++ = 0;
+    
+    char filename[256];
+    uint8_t len = MIN(RAM[FNLEN], sizeof(filename) - 1);
```

Comment:
RAM[FNLEN] is a byte and can't be > 255, so the MAX() part is a no-op. I'd rather comment on the line above that 256 is the maximum possible KERNAL filename length + 1.